### PR TITLE
Upgrade combobox-nav to v2 for ARIA 1.2

### PR DESCRIPTION
**Pull request by muan**
This upgrades `combobox-nav` to v2 for ARIA 1.2 combobox pattern support.

Note that this is a breaking change because we are not using `aria-owns` anymore. I changed it to `for` so it's more neutral and is unlikely to be changed in the future for external reasons.

This also adds optional support for `alt` + arrow keys (https://github.com/github/auto-complete-element/pull/42/commits/3c434192d515f51566ca8d53b37edac24731f3c8). However I wonder what people's opinion on this is? as there's already a default action for alt + arrow keys, which is to move cursor position. Adding support for this means we'd either 1. override the default behavior 2. add this a side-effect for the default behavior. Personally I find 1 not ideal, and 2 unexpected (moving cursor causing the listbox to expand). ~~For now I've opted for 2, but I wonder if folks have strong opinions on this.~~ Opted for 1.

This also removes the `umd` build as we don't have a need for it anymore, changes the `module` build to treat `combobox-nav` as an external dependency, and adds a `browser` build which treats `combobox-nav` as an internal dependency. The `browser` build is required as browsers can't resolve `import '@github/combobox-nav'`. 

[Test page](https://html-is.glitch.me/github/auto-complete-element-42.html). Inputting `@` will match all the results. cc @jscholes

### CODE REVIEW

**Comment by jscholes** [comment]
@muan 

> there's already a default action for alt + arrow keys, which is to move cursor position.

Could you elaborate on this?  This sounds nonstandard and unexpected.

**Comment by muan** [comment]
> Could you elaborate on this? This sounds nonstandard and unexpected.

@jscholes Sorry I wasn't clear. It's macOS specific, but I can't find their docs on text area specific navigation. It works like ctrl + arrow keys on Windows. 

It's in the [mac's Pages app shortcut docs](https://support.apple.com/guide/pages/keyboard-shortcuts-tanc0ffef022/mac) though for what it's worth:

- Move to the beginning of the paragraph | Control-A or Option-Up Arrow
- Move to the end of the paragraph | Control-E or Option-Down Arrow

This works in all text areas across browsers on mac.

**Comment by jscholes** [comment]
Ah, okay.  In that case I'd probably go with option 1, because Mac users can presumably also press FN+Left/Right Arrow to move to the beginning or end of line.  If they don't have an FN key, the text is likely to be short.

It's worth remembering that on the Mac, this pattern isn't common in the first place.  A native `<select>` manifests itself as a "Menu Button" (in ARIA parlance), so users are already encountering something they don't really expect.

**Comment by muan** [comment]
@jscholes Thanks. It seems like Up/Down arrows without modifiers do the same thing inside of an input as well, not text area, but in this case we are only dealing with input I've made the change. Have you found any issues with the test case?

**Comment by jscholes** [comment]
No, this should be good to go.  Looking forward to the accessibility 
improvements this will bring and facilitate!

File: rollup.config.js [hunk 1]
```diff
@@ -5,24 +5,32 @@ import resolve from 'rollup-plugin-node-resolve'
...
+      file: pkg['browser'],
+      format: 'es'
+    },
+    plugins: [
+      resolve(),
+      babel({
+        presets: ['github']
+      })
+    ]
+  }
```
Comment (keithamus) [inline rollup.config.js:35]: > The browser build is required as browsers can't resolve `import '@github/combobox-nav'`.

The point of the `browser` field is to provide an alternative compilation that avoids using Node.js modules which cannot be resolved. As a more demonstrative example of this, the `browser` field can be an object that resolves various packages to alternate stub files; e.g. 

```json
{
  "browser": {
    "fs": "./fs-stub-browser.js"
  }
}
```

The expectation is that you're using a bundler which will rewrite `import` statements. External dependencies are fair game.

GitHub's compiler config respects the `browser` key, and so as this stands we would consume this browser version. We should drop this browser compile step, as it isn't needed.

**Pull request merged**